WordPress/gutenberg
# Query Pagination Previous: Add support for maximum page limit

Fixes: #67748 

## What?
The **Query Pagination Previous** block is displayed even when a user navigates beyond the range of available query results. Since no results are shown in such cases, displaying this block is unnecessary and can be confusing. This PR addresses the issue by ensuring the **Query Pagination Previous** block is hidden when the query has no results, improving user experience and interface clarity.

## How?
The logic has been updated to render the block only when the page number is greater than 1 and less than or equal to the total number of pages, ensuring accurate and appropriate visibility.

## Testing Instructions
1. Add a query loop block.
2. Make sure to add Query Pagination block which contains Query Pagination Previous Block as well.
3. Enter a page range that lies outside the query range.
4. Observe that the block **Query Pagination Previous** is not rendered.

## Screencast

https://github.com/user-attachments/assets/2cf3cc25-215f-4eb6-877d-ac2f6086f7f5

## Screenshots

**Before:**
![Screenshot 2024-12-10 at 11 42 10 AM](https://github.com/user-attachments/assets/38549725-eaa4-40f1-a7b9-528842446cc7)


**After:**
![Screenshot 2024-12-10 at 11 41 08 AM](https://github.com/user-attachments/assets/a339cab2-9a8c-4e00-bd2a-f1bb1ed3b2c4)
